src/write-archive/index.ts

```diff
@@ -98,6 +89,24 @@ export async function writeTestResult(
   }
 }
 
+async function writeSnapshotFiles(
+  domSnapshots: Record<string, Buffer>,
+  archiveDir: string,
+  title: string,
+  sourceMap: Map<string, string>
+) {
+  // The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received an instance of Object
+  await Object.entries(domSnapshots).map(async ([name, domSnapshot]) => {
+    // XXX_jwir3: We go through our stories here and map any instances that are found in
+    //            the keys of the source map to their respective values.
+    const mappedSnapshot = await mapSourceEntries(domSnapshot, sourceMap);
+
+    await outputFile(
+      join(archiveDir, `${sanitize(title)}-${sanitize(name)}.snapshot.json`),
+      mappedSnapshot
+    );
+  });
+}
```

Comment:
I extracted this since we're now calling this from both Playwright and Cypress. It'll be obsolete once the source map [changes](https://github.com/chromaui/test-archiver/pull/28) are merged.